lib/fastlane/actions/xcodebuild.rb

```diff
@@ -316,6 +316,8 @@ class XcbuildAction < Action
       def self.run(params)
         params_hash = params || {}
         params_hash[:build] = true
+
+        SetBuildNumberRepositoryAction.add_build_number_build_setting(params_hash, :build_settings)
```

Comment:
The same here, enforcing to set a build number in xcodebuild action is not a good idea. Myself, I always want to control this in my flow.